proto/outbound.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package io.linkerd.proxy.outbound;
+
+option go_package = "github.com/linkerd/linkerd2-proxy-api/go/outbound";
+
+import "net.proto";
+import "destination.proto";
+import "meta.proto";
+import "http_route.proto";
+
+service OutboundPolicies {
+  rpc Get(TargetSpec) returns (Service) {}
+
+  rpc Watch(TargetSpec) returns (stream Service) {}
+}
+
+message TargetSpec {
+  // Identifies the source proxy workload (e.g., pod name).
+  string workload = 1;
+
+  // Target address
+  io.linkerd.proxy.net.IPAddress address = 2;
+
+  // Target port
+  uint32 port = 3;
+}
+
+message Service {
+  repeated HttpRoute http_routes = 1;
```

Comment:
Is this an `HttpService` Or how are we going to handle other protocols? What do we do on the inbound side?

I think we should probably at least understand how the GrpcRoute will fit in here, but in order to replace TrafficSplit completely we're also going to have to sort out the TCP case... How will that stuff fit in here?

Reply:
More generally, is it part of the API contract that we will only ever lookup _service_ IPs? Might we lookup pod IPs via this API at some point?

I'd probably try to stay in line with whatever we do on the inbound side, but my gut says this should be an _OutboundPolicy_.

Reply:
On the inbound side, we return a ProxyProtocol enum with HTTPRoutes attached to the HTTP and Detect variants.  This is informed by the `proxy_protocol` field on the Server resource.  We don't really have anything equivalent to inform the protocol on the outbound side.  

I guess we kind of treat this `Service` message similarly to `Detect` on the outbound side in that it has a list of http routes which are to be used if the protocol is detected to be HTTP.  Similarly, we could add policy for other protocols here as well which should be used when the detected protocol matches.